### DISCUSSION TITLE
OpenSSL Source Code

### DISCUSSION THREAD

**Discussion by wdkaye** [DominoJP/Image-Encryption-AES#8] [General]
I think I found the part in the open-source OpenSSL repository where they implement the algorithm:

https://github.com/openssl/openssl/blob/master/crypto/aes/aes_core.c

I say this as one who is still inexperienced with AES encryption, and who swears blind allegiance to the perhaps-overly-pedantic [JPL C Coding Standard](https://yurichev.com/mirrors/C/JPL_Coding_Standard_C.pdf), but... eww, this is some imperfect code.  No code comments, magic numbers galore, pre-compile optimizations without explanation, #ifdefs in the middle of functions, ...bleah

**Comment by Andrew-Miner** [comment]
Man that code is rough to read through. Lots of cool stuff in it though. Like the union at the start is a great idea. It would save me from having to do most of the typecasting in my code.
